app/src/main/java/com/github/orkest/ViewModel/profile/ProfileViewModel.kt

```diff
@@ -97,5 +97,102 @@ open class ProfileViewModel(private val user: String) : ViewModel() {
+        userDocument(user).get()
+            .addOnSuccessListener { document ->
+                if (document != null && document.exists()) {
+                    val user = document.toObject(User::class.java)
+                    if (user != null && user.followers.contains(Constants.currentLoggedUser)) {
+                        future.complete(true)
+                    } else{
+                        future.complete(false)
+                    }
+                }
+            }
+            .addOnFailureListener { e -> Log.w(TAG, "Error getting user data", e) }
+        return future
+    }
+
+    open fun follow(): CompletableFuture<Boolean>{
+        val futureFollow =CompletableFuture<Boolean>()
+
+        val userUpdated = updateUser(toFollow = true)
+        val currentUserUpdated = updateCurrentUser(toFollow = true)
+
+        //Both updates must be successful
+        futureFollow.complete(userUpdated && currentUserUpdated)
+        return futureFollow
+    }
+
+    open fun unfollow(): CompletableFuture<Boolean>{
+        val futureFollow = CompletableFuture<Boolean>()
+
+        val userUpdated = updateUser(toFollow = false)
+        val currentUserUpdated = updateCurrentUser(toFollow = false)
+
+        //Both updates must be successful
+        futureFollow.complete(userUpdated && currentUserUpdated)
+        return futureFollow
+    }
+
+    /**
+     * Updates the user's followers' list
+     * toFollow: Boolean = represents whether or not the current logged in user wants to follow this account
+     */
+    private fun updateUser(toFollow: Boolean): Boolean{
+        if(user == Constants.currentLoggedUser){ Log.e(TAG, "Cannot call this function when visiting the current logged-in user's profile", IllegalArgumentException()) }
+        var userUpdated = false
+        userDocument(user).get().addOnSuccessListener { document ->
+                if (document != null && document.exists()) {
+                    val user = document.toObject(User::class.java)
+                    if (user != null) {
+                        if(toFollow){
+                            user.profile.nbFollowers += 1
+                            user.followers.add(Constants.currentLoggedUser)
+                            userUpdated = true
+                        } else {
+                            if (user.profile.nbFollowers > 0) user.profile.nbFollowers -= 1
+                            user.followers.remove(Constants.currentLoggedUser)
+                            userUpdated = true
+                        }
+                    }
+                }
+            }
+            .addOnFailureListener { e -> Log.w(TAG, "Error getting user data", e) }
+        return userUpdated
+    }
+
+    /**
+     * Updates the current logged-in user's followings' list
+     * toFollow: Boolean = represents whether or not the current logged in user wants to follow this account
+     */
+    private fun updateCurrentUser(toFollow: Boolean): Boolean{

```

Comment:
I beleive the name of the method is not very explicit

Reply:
I will update it, thank you